urdf_parser/src/joint.cpp

```diff
@@ -639,46 +642,42 @@ bool exportPose(Pose &pose, TiXmlElement* xml);
 
 bool exportJointDynamics(JointDynamics &jd, TiXmlElement* xml)
 {
-  TiXmlElement *dynamics_xml = new TiXmlElement("dynamics");
-  dynamics_xml->SetAttribute("damping", urdf_export_helpers::values2str(jd.damping) );
-  dynamics_xml->SetAttribute("friction", urdf_export_helpers::values2str(jd.friction) );
-  xml->LinkEndChild(dynamics_xml);
+  auto * dynamics_xml = xmlAppendChild(xml, "dynamics");
```

Comment:
I really prefer not to use `auto` unless it is blindingly obvious from this line of code what the type is.  It doesn't seem to be the case here, so I'd much prefer using `tinyxml2::XMLElement*` here.  The same goes for most uses of `auto` below.